pkg/arvo/gen/hood/tomb.hoon

```diff
@@ -1,32 +1,117 @@
-::  Perform minimal norm change to delete a file, use =dry & for dry run
+::    perform minimal norm change to permanently delete files
 ::
-::  TODO: recognize when it's going to fail because it's in the head of
-::        a desk, and maybe offer to |rm
+::  use =dry & for dry run
 ::
+/+  *generators, *sole
+=,  space:userlib
 =,  clay
-:-  %say
+:-  %ask
 |=  [[now=@da eny=@uvJ bec=beak] [target=path ~] dry=_|]
-:-  %helm-pans
-=+  .^(=arch %cy target)
-?~  fil.arch
-  [%d %flog %text "tomb: not a file"]~  ::  should recurse
-=/  =lobe  u.fil.arch
+|^
 =+  .^(=rang %cx /(scot %p p.bec)//(scot %da now)/rang)
 =+  .^(=cone %cx /(scot %p p.bec)//(scot %da now)/domes)
-=/  domes=(list [[=ship =desk] foam])  ~(tap by cone)
-=/  norms
+=/  =beam  (need (de-beam target))                      ::  beam of target
+=/  dusk=desk  q:beam                                   ::  desk to delete from
+=+  .^(do=dome %cv target)                              ::  dome of that desk
+=/  domes=(list [[=ship =desk] foam])                   ::  all domes on ship
+  ~(tap by cone)
+=+  .^(=cass %cw (en-beam beam(r da+now)))              ::  latest aeon
+=/  used=(list [desk path])                             ::  desks using the target
+  =-  (murn lobes -)                                    ::  match over lobes
+  |=  lob=lobe
+  ^-  (unit [desk path])
+  =/  doms=(list [[=ship =desk] foam])  domes
```

Comment:
I don't think you need this `doms` local variable, or the `|-` below it.  You should be able to use the `|=  lob=lobe` gate as the outer recursion point.

Reply:
nvm I'm wrong